ipi/engine/normalmodes.py

```diff
@@ -700,24 +736,27 @@ def get_dynm3(self):
 
     def get_vspring_and_fspring_B(self):
         """
-        Calculates spring forces and potential for bosons.
-        Evaluated using recursion relation from arXiv:1905.090.
+        Calculates spring forces and potential, including exchange effects for the bosons present, if any.
```

Comment:
I would restructure this quite heavily. Rather than having a joing v_and_f for bosons, and separate v and f for distinguishable particles, I'd have a single v_and_f for everything, that uses bosons if needed or normal modes otherwise. then you can have a pair of v and f depends that just separate the two components. you can keep the two functions separate if you prefer, but I think at the level of depend objects that would be a better structure.